ComputeAdmin/AzureStack.ComputeAdmin.psm1

```diff
@@ -660,5 +662,185 @@ Function CreateGalleyItem{
+    }
+    
+    $osTypeJSON = '"VmOsType" : "' + $osType + '"'
+    $ComputeRoleJSON = '"ComputeRole" : "N/A"'
+    $VMScaleSetEnabledJSON = '"VMScaleSetEnabled" : "' + $vmScaleSetEnabled + '"'
+    $SupportMultipleExtensionsJSON = '"SupportMultipleExtensions" : "' + $supportMultipleExtensions + '"'
+    $IsSystemExtensionJSON = '"IsSystemExtension" : "false"'
+
+    $propertyBody = $sourceBlobJSON + "," + $osTypeJSON + ',' + $ComputeRoleJSON + "," + $VMScaleSetEnabledJSON + "," + $SupportMultipleExtensionsJSON + "," + $IsSystemExtensionJSON 
+
+    #building ARMResource
+
+    $RequestBody = '{"Properties":{'+$propertyBody+'}}'
+
+    Invoke-RestMethod -Method PUT -Uri $uri -Body $RequestBody -ContentType 'application/json' -Headers $Headers
+
+    $extensionHandler = Invoke-RestMethod -Method GET -Uri $uri -ContentType 'application/json' -Headers $Headers
+
+    while($extensionHandler.Properties.ProvisioningState -ne 'Succeeded')
+    {
+        if($extensionHandler.Properties.ProvisioningState -eq 'Failed')
+        {
+            Write-Error -Message "VM extension download failed." -ErrorAction Stop
+        }
+
+        if($extensionHandler.Properties.ProvisioningState -eq 'Canceled')
+        {
+            Write-Error -Message "VM extension download was canceled." -ErrorAction Stop
+        }
+
+        Write-Host "Downloading";
+        Start-Sleep -Seconds 4
+        $extensionHandler = Invoke-RestMethod -Method GET -Uri $uri -ContentType 'application/json' -Headers $Headers
+    }
+
+    Remove-AzureStorageContainer –Name $containerName -Force
+    Remove-AzureRmStorageAccount -ResourceGroupName $resourceGroupName -AccountName $storageAccountName 
+    Remove-AzureRmResourceGroup -Name $resourceGroupName -Force
+}
+
+Function Remove-VMExtension{
+    Param(
+        [ValidatePattern(“[a-zA-Z0-9-]{3,}”)]
+        [String] $publisher,
+
+        [Parameter(Mandatory=$true)]
+        [ValidatePattern(“\d+\.\d+\.\d+”)]
+        [String] $version,
+
+        [String] $type = "CustomScriptExtension",
+
+        [Parameter(Mandatory=$true)]
+        [ValidateSet('Windows' ,'Linux')]
+        [String] $osType,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullorEmpty()]
+        [String] $tenantID,
+
+        [String] $location = 'local',
+
+        [Parameter(Mandatory=$true)]
+        [System.Management.Automation.PSCredential] $azureStackCredentials,
+
+        [string] $ArmEndpoint = 'https://adminmanagement.local.azurestack.external'
+    )
+
+    $subscription, $headers =  (Get-AzureStackAdminSubTokenHeader -TenantId $tenantId -AzureStackCredentials $azureStackCredentials -ArmEndpoint $ArmEndpoint)
+
+    $ArmEndpoint = $ArmEndpoint.TrimEnd("/")
+    $uri = $armEndpoint + '/subscriptions/' + $subscription + '/providers/Microsoft.Compute.Admin/locations/' + $location + '/artifactTypes/VMExtension/publishers/' + $publisher
+    $uri = $uri + '/types/' + $type + '/versions/' + $version + '?api-version=2015-12-01-preview'
```

Comment:
No worry. Seems this happens all everywhere else in this project, keep it as is now and we can refactor all at once